core/config/config.go

```diff
@@ -166,8 +166,9 @@ func (c Config) GetMaxMemoryFactor() (float64, error) {
 
 func (c Config) HaveEthClient() bool {
 	if c.Blockchain.ChainProvider == blockchain.ProviderNullChain {
-		return false
+		return c.IsValidator() && len(c.Ethereum.RPCEndpoint) > 0
 	}
+
 	return c.IsValidator()
 }
```

Comment:
I wonder if this function shouldn't be simply:
```go
func (c Config) HaveEthClient() bool {
	return c.IsValidator() && len(c.Ethereum.RPCEndpoint) > 0
}
```

Because, being a validator without Ethereum RPC endpoint set should be an error, no ?

Reply:
fixed